test/SparqlAntlrParserTestHelpers.h

```diff
@@ -327,33 +344,36 @@ MATCHER_P4(IsSolutionModifier, groupByVariables, havingClauses, orderBy,
 
 // TODO: ExplainMatchResult?
 MATCHER_P(IsTriples, triples, "") {
-  return testing::Matches(testing::UnorderedElementsAreArray(triples))(
-      std::get<GraphPatternOperation::BasicGraphPattern>(arg.variant_)
-          ._whereClauseTriples);
-}
-
-MATCHER_P2(IsBindd, target, expression, "") {
-  auto& bind = std::get<GraphPatternOperation::Bind>(arg.variant_);
-  return bind._target == target &&
-         bind._expression.getDescriptor() == expression;
-}
-
-MATCHER_P(IsFilters, filters, "") {
-  return testing::Matches(testing::UnorderedElementsAreArray(filters))(
-      arg._filters);
+  auto triplesValue =
+      std::get_if<GraphPatternOperation::BasicGraphPattern>(&arg.variant_);
+  return triplesValue && testing::Matches(testing::UnorderedElementsAreArray(
+                             triples))(triplesValue->_whereClauseTriples);
 }
 
-MATCHER(IsOptional, "") {
-  return testing::Matches(testing::Eq(true))(arg._optional);
+MATCHER_P(IsOptional, subMatcher, "") {
+  auto optional = std::get_if<GraphPatternOperation::Optional>(&arg.variant_);
+  return optional && testing::Value(optional->_child, subMatcher);
 }
 
-MATCHER_P(IsChildren, childMatchers, "") {
-  if (arg._children.size() != childMatchers.size()) {
+MATCHER_P3(IsGraphPattern, optional, filters, childMatchers, "") {
+  if (arg._children.size() != std::tuple_size_v<decltype(childMatchers)>) {
     return false;
   }
-  bool matches = true;
-  for (size_t i = 0; i < childMatchers.size(); ++i) {
-    matches &= testing::Matches(childMatchers[i])(arg._children[i]);
-  }
-  return matches;
-}
+
+  auto sequence =
+      std::make_index_sequence<std::tuple_size_v<decltype(childMatchers)>>();
+
+  auto lambda = [&]<size_t... i>(std::index_sequence<i...>,
+                                 auto&&... matchers) {
+    return (... && testing::Value(arg._children[i], matchers));
+  };
+  auto lambdalambda = [&lambda, &sequence](auto... matchers) {
+    return lambda(sequence, matchers...);
+  };
+  bool childrenMatch = std::apply(lambdalambda, childMatchers);
+
```

Comment:
I have Ideas on how to make this more readabele that I wyould like to discuss in person.